pkg/util/util.go

```diff
@@ -239,3 +244,181 @@ func SplitQualifiedName(str string) (string, string) {
+			break
+		}
+		if strings.HasPrefix(line, "Iface") {
+			continue
+		}
+		fields := strings.Fields(line)
+		routes = append(routes, Route{})
+		route := &routes[len(routes)-1]
+		route.Interface = fields[0]
+		ip, err := ParseIP(fields[1])
+		if err != nil {
+			return nil, fmt.Errorf("Unable to Parse IP: %s", fields[1])
+		}
+		route.Destination = ip
+		ip, err = ParseIP(fields[2])
+		if err != nil {
+			return nil, fmt.Errorf("Unable to Parse IP: %s", fields[2])
+		}
+		route.Gateway = ip
+	}
+	return routes, nil
+}
+
+func ParseIP(str string) (net.IP, error) {
+	bytes, err := hex.DecodeString(str)
+	if err != nil {
+		return nil, err
+	}
+	//TODO add ipv6 support
+	if len(bytes) != net.IPv4len {
+		return nil, fmt.Errorf("only IPv4 is supported")
+	}
+	bytes[0], bytes[1], bytes[2], bytes[3] = bytes[3], bytes[2], bytes[1], bytes[0]
+	return net.IP(bytes), nil
+}
+
+func isInterfaceUp(intf *net.Interface) bool {
+	if intf.Flags&net.FlagUp != 0 {
+		glog.V(4).Infof("Interface %v is up", intf.Name)
+		return true
+	}
+	return false
+}
+
+func getFinalIP(addrs []net.Addr) (net.IP, error) {
+	if len(addrs) > 0 {
+		for i := range addrs {
+			glog.V(4).Infof("Checking addr  %s.", addrs[i].String())
+			ip, _, err := net.ParseCIDR(addrs[i].String())
+			if err != nil {
+				return nil, err
+			}
+			//Only IPv4
+			//TODO : add IPv6 support
+			if ip.To4() != nil {
+				if !ip.IsLoopback() {
+					glog.V(4).Infof("IP found %v", ip)
+					return ip, nil
+				} else {
+					glog.V(4).Infof("Loopback found %v", ip)
+				}
+			} else {
+				glog.V(4).Infof("%v is not a valid IPv4 address", ip)
+			}
+
+		}
+	}
+	return nil, nil
+}
+
+func getIPFromInterface(intfName string, nw NwInterface) (net.IP, error) {
+	intf, err := nw.InterfaceByName(intfName)
+	if err != nil {
+		return nil, err
+	}
+	if isInterfaceUp(intf) {
+		addrs, err := nw.Addrs(intf)
+		if err != nil {
+			return nil, err
+		}
+		glog.V(4).Infof("Interface %q has %d addresses :%v.", intfName, len(addrs), addrs)
+		finalIP, err := getFinalIP(addrs)
+		if err != nil {
+			return nil, err
+		}
+		if finalIP != nil {
+			glog.V(4).Infof("valid IPv4 address for interface %q found as %v.", intfName, finalIP)
+			return finalIP, nil
+		}
+	}
+
+	return nil, nil
+}
+
+//ChooseHostInterface is a method used fetch an IP for a daemon.
+//It uses data from /proc/net/route file.
+//For a node with no internet connection ,it returns error
+//For a multi n/w interface node it returns the IP of the interface with gateway on it.
+func ChooseHostInterface() (net.IP, error) {
+	var finalIP net.IP
+	inFile, err := os.Open("/proc/net/route")
+	if err != nil {
+		return nil, err
+	}
+	defer inFile.Close()
+	routes, err := getRoutes(inFile)
+	if err != nil {
+		fmt.Printf("getRoutes, %v", err)
+		return nil, err
+	}
+	var nw NwInterface = NetworkInterface{}
+	finalIP, err = chooseHostInterfaceFromRoute(routes, nw)
+	if err != nil {
+		return nil, err
+	}
+	return finalIP, err
+}
+
+type NwInterface interface {
```

Comment:
type networkInterfacer

It doesn't need to be exported and the "er" suffix is go convention